apollo-core/src/main/java/com/spotify/apollo/core/ServiceImpl.java

```diff
@@ -207,12 +214,27 @@ InstanceImpl initInstance(
       ListeningScheduledExecutorService scheduledExecutorService,
       CountDownLatch shutdownRequested,
       CountDownLatch stopped) {
-      List<ApolloModule> modulesSortedOnPriority =
+
+    List<ApolloModule> modulesSortedOnPriority =
               modules.stream().sorted(Ordering.natural().reverse().onResultOf(ModulePriorityOrdering.INSTANCE))
                      .collect(Collectors.toList());
 
     Iterable<Module> allModules = concat(of(coreModule), modulesSortedOnPriority);
-    Injector injector = Guice.createInjector(Stage.PRODUCTION, allModules);
+
+    final SpringApplicationBuilder springApplicationBuilder = new SpringApplicationBuilder()
+        .properties(
+           //Prefer Spring bean if present in both Guice and application context
+            "spring.guice.dedup=true",
+           //Do not create default instances if there is no instance in the context
+           "spring.guice.autowireJIT=false",
+          //Allow overriding of bean definitions from different modules
+           "spring.main.allow-bean-definition-overriding=true")
+        .sources(GuiceSpringBridge.class)
```

Comment:
Since you are adding this class as a source I am fairly certain you can leverage the @ComponentScan annotation on this class to enable picking up @Configuration, @Bean, and all variations of @Component